modules/rostests/apitests/win32nt/ntuser/NtUserCreateAcceleratorTable.c

```diff
@@ -0,0 +1,188 @@
+    {
+        hAccel = NtUserCreateAcceleratorTable(NULL, MIN_VALID_NUMBER);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        bHung = TRUE;
+    }
+    _SEH2_END;
+
+    ok_long(GetLastError(), 0xdeadbeef);
+    ok_int(bHung, FALSE);
+    ok_hdl(hAccel, NULL);
+
+    /* Try invalid Entries argument */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    _SEH2_TRY
+    {
+        hAccel = NtUserCreateAcceleratorTable((LPACCEL)(ULONG_PTR)0xC0FEC0FE, MIN_VALID_NUMBER);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        bHung = TRUE;
+    }
+    _SEH2_END;
+
+    ok_long(GetLastError(), ERROR_NOACCESS);
+    ok_int(bHung, FALSE);
+    ok_hdl(hAccel, NULL);
+
+    /* Try EntriesCount = 0 */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    _SEH2_TRY
+    {
+        hAccel = NtUserCreateAcceleratorTable(Entries, MIN_VALID_NUMBER - 1);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        bHung = TRUE;
+    }
+    _SEH2_END;
+
+    ok_long(GetLastError(), 0xdeadbeef);
+    ok_int(bHung, FALSE);
+    ok_hdl(hAccel, NULL);
+
+    /* Try minimum */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    _SEH2_TRY
+    {
+        hAccel = NtUserCreateAcceleratorTable(Entries, MIN_VALID_NUMBER);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        bHung = TRUE;
+    }
+    _SEH2_END;
+
+    ok_long(GetLastError(), 0xdeadbeef);
+    ok_int(bHung, FALSE);
+    ok(hAccel != NULL, "hAccel is NULL\n");
+
+    if (!bHung && hAccel != NULL)
+        DestroyAcceleratorTable(hAccel);
+
+    /* Try correct parameters */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    _SEH2_TRY
+    {
+        hAccel = NtUserCreateAcceleratorTable(Entries, _countof(Entries));
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        bHung = TRUE;
+    }
+    _SEH2_END;
+
+    ok_long(GetLastError(), 0xdeadbeef);
+    ok_int(bHung, FALSE);
+    ok(hAccel != NULL, "hAccel is NULL\n");
+
+    if (!bHung && hAccel != NULL)
+        DestroyAcceleratorTable(hAccel);
+
+    /* Try maximum */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    pEntries = HeapAlloc(GetProcessHeap(), 0, MAX_VALID_NUMBER * sizeof(*pEntries));
+    if (pEntries == NULL)
+    {
+        skip("pEntries is NULL\n");
+    }
+    else
+    {
+        _SEH2_TRY
+        {
+            hAccel = NtUserCreateAcceleratorTable(pEntries, MAX_VALID_NUMBER);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            bHung = TRUE;
+        }
+        _SEH2_END;
+
+        ok_long(GetLastError(), 0xdeadbeef);
+        ok_int(bHung, FALSE);
+        ok(hAccel != NULL, "hAccel is NULL\n");
+
+        if (!bHung && hAccel != NULL)
+            DestroyAcceleratorTable(hAccel);
+
+        HeapFree(GetProcessHeap(), 0, pEntries);
+    }
+
+    /* Try maximum +1 */
+    SetLastError(0xdeadbeef);
+    bHung = FALSE;
+    pEntries = HeapAlloc(GetProcessHeap(), 0, (MAX_VALID_NUMBER + 1) * sizeof(*pEntries));
```

Comment:
Ah, I expected this case to fail.
Could you find the real limit (if any) out?

Reply:
Windows failed if test try allocate table with MAX_VALID_NUMBER + 57700000.